src/Bicep.LangServer/Handlers/BicepDecompileSaveCommandHandler.cs

```diff
@@ -0,0 +1,274 @@
+                if (!overwrite)
+                {
+                    Debug.Assert(filesToSave.All(f => IsPathRelativeToBasePath(f.path, outputFolder)), $"Expected all output files to be relative to {outputFolder}");
+                }
+                var actualMainBicepPath = filesToSave[0].path;
+
+                // Save files
+                SaveFiles(output, filesToSave);
+
+                // Completion message
+                Log(output, $"Decompilation complete.");
+
+                // Show main output file
+                if (this.clientCapabilitiesProvider.DoesClientSupportShowDocumentRequest())
+                {
+                    await this.languageServerFacade.Window.ShowDocument(
+                        new() { TakeFocus = true, Uri = actualMainBicepPath },
+                        CancellationToken.None);
+                }
+
+                // Return result
+                return (
+                    result: new BicepDecompileSaveCommandResult(
+                        output.ToString(),
+                        null,
+                        actualMainBicepPath,
+                        filesToSave.Select(f => f.path).ToArray()),
+                    successTelemetry: BicepTelemetryEvent.DecompileSaveSuccess(decompileId)
+                    );
+            }
+            catch (Exception ex)
+            {
+                Log(output, ex.Message);
+                throw telemetryHelper.CreateException(
+                    ex.Message,
+                    BicepTelemetryEvent.DecompileSaveFailure(decompileId, ex.GetType().Name),
+                    new BicepDecompileSaveCommandResult(
+                        output.ToString(),
+                        ex.Message,
+                        null,
+                        new string[] { })
+                );
+            }
+        }
+
+        private (string path, string contents)[] DeterminePathsToSave(StringBuilder output, ref string outputFolder, string mainBicepPath, DecompiledFile[] outputFiles, bool overwrite)
+        {
+            var singleFileDecompilation = outputFiles.Length == 1;
+
+            if (overwrite)
+            {
+                // Save to original paths
+                return outputFiles.Select(of => (of.absolutePath, of.bicepContents)).ToArray();
+            }
+
+            if (singleFileDecompilation)
+            {
+                // Create a bicep file with unique name alongside the existing bicep file
+                string newBicepPath = FindUniqueFileOrFolderName(outputFolder, mainBicepPath);
+                return new[] { (newBicepPath, outputFiles[0].bicepContents) };
+            }
+            else
+            {
+                // Output every to a new subfolder (using the relative clonable paths)
+                var newOutputFolder = CreateUniqueSubfolder(outputFolder, $"{Path.GetFileNameWithoutExtension(mainBicepPath)}_decompiled");
```

Comment:
Can we just use outputFolder instead of creating newOutputFolder?

Reply:
[sw/decompile2 d9b3c902b] CR fixes